vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/dataview/CheckboxGroupListDataViewImpl.java

```diff
@@ -0,0 +1,152 @@
+        CheckboxGroupListDataView<T> dataView = withFilterOrOrder(
+                dataProvider -> dataProvider.setFilter(filter));
+        // fire size change event if size has changed
+        getDataSize();
+        return dataView;
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> withSortComparator(SerializableComparator<T> sortComparator) {
+        return withFilterOrOrder(dataProvider -> dataProvider.setSortComparator(sortComparator));
+    }
+
+    @Override
+    public Stream<T> getAllItems() {
+        DataProvider<T, ?> dataProvider = getDataProvider();
+        return dataProvider.fetch(new Query<>());
+    }
+
+    @Override
+    public int getDataSize() {
+        int size = getDataProvider().size(new Query<>());
+        if (size != itemsSize && sizeChangeListeners != null) {
+            sizeChangeListeners.forEach(listener -> listener.sizeChanged(
+                    new SizeChangeEvent<>(this.component, size)));
+            itemsSize = size;
+        }
+        return size;
+    }
+
+    @Override
+    public boolean dataContainsItem(T item) {
+        // InMemoryDataProviders do not override getId(), so rely on equals() method
+        return getAllItems().collect(Collectors.toList()).contains(item);
+    }
+
+    @Override
+    public T getItemOnRow(int row) {
+        if (row < 0) {
+            throw new IndexOutOfBoundsException("Row number should be zero on greater");
+        }
+
+        List<T> itemsList = getAllItems().collect(Collectors.toList());
+
+        if (itemsList.isEmpty()) {
+            throw new IndexOutOfBoundsException("Item requested on an empty data set");
+        }
+
+        final int itemsSize = itemsList.size();
+        if (row >= itemsSize) {
+            throw new IndexOutOfBoundsException(
+                    String.format("Row number is outside of data set size, row=%d, data set size=%d", row, itemsSize));
+        }
+
+        return itemsList.get(row);
+    }
+
+    // This could be a common method on top of List- and Lazy- DataViews
+    @Override
+    public Registration addSizeChangeListener(SizeChangeListener listener) {
+        if (sizeChangeListeners == null) {
+            sizeChangeListeners = new HashSet<>();
+        }
+        sizeChangeListeners.add(listener);
+        return () -> sizeChangeListeners.remove(listener);
+    }
+
+    // This is specific for components
+    private DataProvider<T, ?> getDataProvider() {
+        return this.component.getDataProvider();
+    }
+
+    @SuppressWarnings("unchecked")
+    private CheckboxGroupListDataView<T> withFilterOrOrder(
+            SerializableConsumer<ListDataProvider<T>> filterOrOrderConsumer) {
+        DataProvider<T, ?> dataProvider = getDataProvider();
+        filterOrOrderConsumer.accept((ListDataProvider<T>) dataProvider);
+        return this;
+    }
+
+    private T getNextOrPreviousItem(T item, SerializableFunction<List<T>, T> itemProvider) {
```

Comment:
This doesn't actually provide next nor previous item and is a bit extra as the next and previous methods just need to have the get the item list from the dataprovider.

Also item is not used by this method at all.

Reply:
Removed.